framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -1527,12 +1697,38 @@ public AnnotationMirror convertArrayLenToArrayLenRange(AnnotationMirror arrayLen
         return createArrayLenRangeAnnotation(Collections.min(values), Collections.max(values));
     }
 
+    /** Converts an {@code @StringVal} annotation to an {@code @ArrayLenRange} annotation. */
+    public AnnotationMirror convertStringValToArrayLenRange(AnnotationMirror stringValAnno) {
+        List<String> values = getStringValues(stringValAnno);
+        List<Integer> lengths = ValueCheckerUtils.getLengthsForStringValues(values);
+        return createArrayLenRangeAnnotation(Collections.min(lengths), Collections.max(lengths));
+    }
+
+    /** Converts an {@code @StringVal} annotation to an {@code @ArrayLen} annotation. */
+    public AnnotationMirror convertStringValToArrayLen(AnnotationMirror stringValAnno) {
+        List<String> values = getStringValues(stringValAnno);
+        return createArrayLenAnnotation(ValueCheckerUtils.getLengthsForStringValues(values));
+    }
+
```

Comment:
Currently the methods are used to convert to the required type in `leastUpperBound`. The incoming `StringVal` should already have a limited number of values, so there would be no need to chose `ArrayLenRange` over `ArrayLen`. It would add an additional conversion step (from `StringVal` to `ArrayLen` and then to `ArrayLenRange`) when doing a lub of  `StringVal` and `ArrayLenRange`.
I don't do a check everytime these are called.